core/pom.xml

```diff
@@ -119,7 +119,13 @@
 
         <dependency>
             <groupId>org.codehaus.groovy</groupId>
-            <artifactId>groovy-all</artifactId>
+            <artifactId>groovy</artifactId>
```

Comment:
TL;DR: I think this change is ok, but here is my reasoning...

What are the implications of switching to `groovy` rather than `groovy-all`? I don't trust groovy to tell us when its dependencies are missing until we actually execute the code (even if our groovy code compiles)!

Will the Brooklyn web-console's groovy console still behave the same as it did before? (or might dependencies be missing that a user was making use of previously - e.g. `groovy-datetime` etc.)

Looking in karaf/features/src/main/feature/feature.xml, I see that is unchanged and still has:
```
<bundle dependency="true">mvn:org.codehaus.groovy/groovy-all/${groovy.version}</bundle>
```

Does that mean this change doesn't actually change the behaviour in Karaf? Or will the package-imports of brooklyn-core now be very different?

Looking in the existing brooklyn-master at brooklyn-core's `bundle:headers` to see what it imports for groovy, I see:
```
groovy.lang;version="[2.4,3)",
groovy.time;version="[2.4,3)",
groovy.util;version="[2.4,3)",
org.codehaus.groovy.runtime;version="[2.4,3)",
```

I checked in with `jar -tf ~/.m2/repository/org/codehaus/groovy/groovy/2.4.4/groovy-2.4.4.jar` and I see this has all four of the packages listed above. So I think it's ok.

Similarly, eyeballing the brooklyn-master Karaf's:
```
package:imports | grep groovy 
```
It looks ok - I didn't see any imports that are not in `org.codehaus.groovy:groovy`.